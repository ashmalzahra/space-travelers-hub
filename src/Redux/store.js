import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './RocketsSlice';
import missionsSlice from './MissionsSlice';
// import missionsReducer from "./MissionsSlice";

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
    missions: missionsSlice.reducer,
  },
});

export default store;
