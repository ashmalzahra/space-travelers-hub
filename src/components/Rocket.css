.container {
  display: flex;
  flex-direction: row;
  margin-top: 4%;
  margin-bottom: 4%;
}

.pic {
  max-width: 275px;
}

.title {
  margin-left: 2.5%;
}

.description {
  margin-left: 2.5%;
  font-size: 20px;
}

.btn-primary,
.btn-light {
  margin-left: 2.5%;
}

.btn-success {
  margin-right: 1.5%;
}
