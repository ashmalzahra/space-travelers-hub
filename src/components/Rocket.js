import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import Button from 'react-bootstrap/Button';
import './Rocket.css';
import { reserveRocket, cancelReserveRocket } from '../Redux/RocketsSlice';

const Rocket = ({ rocket }) => {
  const dispatch = useDispatch();

  const reserveRocketHandler = (id) => {
    dispatch(reserveRocket(id));
  };

  const cancelRocketHandler = (id) => {
    dispatch(cancelReserveRocket(id));
  };

  return (
    <div className="container" id={rocket.rocket_id}>
      <img src={rocket.flickr_images} alt={rocket.rocket_name} className="pic" />
      <div>
        <h3 className="title">{rocket.rocket_name}</h3>
        <p className="description">
          {rocket.reserved === true && <Button variant="success">Reserved</Button>}
          {rocket.description}
        </p>
        {rocket.reserved === true ? (<Button variant="light" id={rocket.rocket_id} onClick={() => cancelRocketHandler(rocket.rocket_id)}>Cancel Reservation</Button>)
          : (<Button variant="primary" id={rocket.rocket_id} onClick={() => reserveRocketHandler(rocket.rocket_id)}>Reserve Rocket</Button>)}
      </div>
    </div>
  );
};

Rocket.propTypes = {
   rocket: PropTypes.shape({
    rocket_id: PropTypes.string.isRequired,
    rocket_name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    reserved: PropTypes.bool,
  }),
};

export default Rocket;
