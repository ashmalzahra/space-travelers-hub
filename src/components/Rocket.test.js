import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../Redux/store';
import Rocket from './Rocket';

const obj = {
  rocket_name: 'Falcon 1',
  rocket_id: 'falcon1',
};

test('Rocket renders as expected', () => {
  const configureStore = store;
  const component = renderer.create(
    <Provider store={configureStore}>
      <MemoryRouter>
        <Rocket key={obj.rocket_id} rocket={obj} />
      </MemoryRouter>
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

// test('Missions renders correctly', () => {
//   const store = configureStore();
//   const component = renderer.create(
//     <Provider store={store}>
//       <MemoryRouter>
//         <Missions />
//       </MemoryRouter>
//     </Provider>,
//   );
//   const tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
// });

// test('Rockets renders correctly', () => {
//   const store = configureStore();
//   const component = renderer.create(
//     <Provider store={store}>
//       <MemoryRouter>
//         <Rockets />
//       </MemoryRouter>
//     </Provider>,
//   );
//   const tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
// });
