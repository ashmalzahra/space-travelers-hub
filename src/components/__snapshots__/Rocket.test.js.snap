// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket renders as expected 1`] = `
<div
  className="container"
  id="falcon1"
>
  <img
    alt="Falcon 1"
    className="pic"
  />
  <div>
    <h3
      className="title"
    >
      Falcon 1
    </h3>
    <p
      className="description"
    />
    <button
      className="btn btn-primary"
      disabled={false}
      id="falcon1"
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;
